internal/app/app.go

```diff
@@ -0,0 +1,89 @@
+package app
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/metal-toolbox/conditionorc/internal/model"
+	"github.com/spf13/viper"
+
+	runtime "github.com/banzaicloud/logrus-runtime-formatter"
+	"github.com/sirupsen/logrus"
+)
+
+// Config holds configuration data when running mctl
+// App holds attributes for the mtl application
+type App struct {
+	// Viper loads configuration parameters.
+	v *viper.Viper
+	// The kind of application - server/orchestrator
+	AppKind model.AppKind
+	// Sync waitgroup to wait for running go routines on termination.
+	SyncWG *sync.WaitGroup
```

Comment:
this doesn't seem to be used and feels weird to export maybe it can be internal only?

Reply:
I'd like to keep this exported, for this first MVP, the orchestrator isn't [processing events concurrently](https://github.com/metal-toolbox/conditionorc/blob/mvp/internal/orchestrator/orchestrator.go#L83).
 In the next set of changes the `SyncWG` var will be passed to the `orchestrator` so it processes events in separate routines, and we'll have a `app.SyncWG.Wait()` before orc returns on a TERM signal.

Reply:
These are good points, for now I've removed the exported  app `SyncWG` field completely and I'll limit its use in the manner suggested above once it becomes necessary for the `orchestrator` component.